File: backend/dotnet-webapi/src/WorldCup.Api/Models/Match.cs
namespace WorldCup.Api.Models;

public class Match
{
    public string Id { get; set; } = string.Empty;
    public string HomeTeamId { get; set; } = string.Empty;
    public string AwayTeamId { get; set; } = string.Empty;
    public string CityId { get; set; } = string.Empty;
    public string Kickoff { get; set; } = string.Empty;
    public string GroupName { get; set; } = string.Empty;
    public int MatchDay { get; set; }
    public double TicketPrice { get; set; } = 100.0;

    // Navigation properties
    public City City { get; set; } = null!;
    public Team HomeTeam { get; set; } = null!;
    public Team AwayTeam { get; set; } = null!;
}

File: backend/java-spring/src/main/java/com/unosquare/worldcup/model/Match.java
package com.unosquare.worldcup.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "matches")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Match {

    @Id
    private String id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "home_team_id")
    private Team homeTeam;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "away_team_id")
    private Team awayTeam;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "city_id")
    private City city;

    private LocalDateTime kickoff;

    @Column(name = "group_name")
    private String groupName;

    @Column(name = "match_day")
    private int matchDay;

    @Column(name = "ticket_price")
    private Double ticketPrice;
}
